src/Ocelot/ServiceDiscovery/Providers/PolingConsulServiceDiscoveryProvider.cs

```diff
@@ -15,42 +15,109 @@ namespace Ocelot.ServiceDiscovery.Providers
 {
     public class PollingConsulServiceDiscoveryProvider : IServiceDiscoveryProvider
     {
+        private readonly ConsulRegistryConfiguration _config;
         private readonly IOcelotLogger _logger;
-        private readonly IServiceDiscoveryProvider _consulServiceDiscoveryProvider;
-        private readonly Timer _timer;
-        private bool _polling;
+        private readonly IConsulClient _consul;
+        private const string VersionPrefix = "version-";
         private List<Service> _services;
-        private string _keyOfServiceInConsul;
 
-        public PollingConsulServiceDiscoveryProvider(int pollingInterval, string keyOfServiceInConsul, IOcelotLoggerFactory factory, IServiceDiscoveryProvider consulServiceDiscoveryProvider)
-        {;
+        private int _pollingInterval;
+        private ulong _waitIndex;
+
+        public PollingConsulServiceDiscoveryProvider(int pollingInterval, ConsulRegistryConfiguration config, IOcelotLoggerFactory factory, IConsulClientFactory clientFactory)
+        {
+            ;
+            _pollingInterval = pollingInterval == 0 ? 10000 : pollingInterval;
```

Comment:
The default pollinginterval is set by ServiceProviderConfigurationCreator.cs as 0, not sure we want to override it here as that can get confusing!

Reply:
>In addition to index, endpoints that support blocking will also honor a wait parameter specifying a maximum duration for the blocking request. This is limited to 10 minutes. If not set, the wait time defaults to 5 minutes. This value can be specified in the form of "10s" or "5m" (i.e., 10 seconds or 5 minutes, respectively). A small random amount of additional wait time is added to the supplied maximum wait time to spread out the wake up time of any concurrent requests. This adds up to wait / 16 additional time to the maximum duration.

2000 may be too little? I'll set the default pollinginterval in ServiceProviderConfigurationCreator.cs